File: pset1/prob/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    // prompt dollar from user

    float d;
    do
    {
        d = get_float("Change owed: ");
    }
    while (d < 0);

    // change dollar to penny(also rounding off)

    float cent = d * 100;

    int c = round(cent);

    printf("%i\n", c);

    //greedy algorithm

    int k;
    int quater;
    int dime;
    int nickel;
    int penny;

    for (quater = c; quater >= 25; quater -= 25)
    {
        k = k + 1;
    }

    for (dime = quater; dime >= 10; dime -= 10)
    {
        k = k + 1;
    }

    for (nickel = dime; nickel >= 5; nickel -= 5)
    {
        k = k + 1;
    }

    for (penny = nickel; penny >= 1; penny -= 1)
    {
        k = k + 1;
    }

    //print coins

    printf("%i\n", k);
}

File: pset6/prob/cash.py
from cs50 import get_float

# get change owed
while True:
    cash = get_float("Change owed: ")
    if cash > 0:
        break

# change to pennies
penny = round(cash * 100)

# greedy algorithm

coins = 0

# quater
while penny >= 25:
    penny -= 25
    coins += 1

# dime
while penny >= 10:
    penny -= 10
    coins += 1

# nickel
while penny >= 5:
    penny -= 5
    coins += 1

# penny
while penny >= 1:
    penny -= 1
    coins += 1

print(coins)